diff/walking/differ.go

```diff
@@ -55,6 +56,26 @@ func NewWalkingDiff(store content.Store) diff.Comparer {
 	}
 }
 
+func readonlyOverlay(opt []string) []string {
+	out := make([]string, 0, len(opt))
+	upper := ""
+	for _, o := range opt {
+		if strings.HasPrefix(o, "upperdir=") {
+			upper = strings.TrimPrefix(o, "upperdir=")
+		} else if !strings.HasPrefix(o, "workdir=") {
+			out = append(out, o)
+		}
+	}
+	if upper != "" {
+		for i, o := range out {
+			if strings.HasPrefix(o, "lowerdir=") {
+				out[i] = "lowerdir=" + upper + ":" + strings.TrimPrefix(o, "lowerdir=")
```

Comment:
Is it simpler to just append the upper instead of prepending?

Reply:
Order of the dirs in `lowerdir=` option does matter. Leftmost dir represents the top layer of the mount. See [multiple-lower-layers](https://www.kernel.org/doc/html/latest/filesystems/overlayfs.html#multiple-lower-layers).